flink-kubernetes/src/test/java/org/apache/flink/kubernetes/highavailability/KubernetesStateHandleStoreTest.java

```diff
@@ -725,4 +742,157 @@ public void testRemoveAllHandles() throws Exception {
+                                            Exception.class,
+                                            () -> {
+                                                store.releaseAndTryRemove(key);
+                                            });
+                            assertThat(exception, FlinkMatchers.containsCause(discardException));
+
+                            // Now we should see that the node is "soft-deleted". This means it can
+                            // no longer be accessed by the get methods, but the underlying state
+                            // still exists.
+                            assertThat(store.getAllAndLock().size(), is(0));
+                            assertThrows(Exception.class, () -> store.getAndLock(key));
+                            assertThat(TestingLongStateHandleHelper.getGlobalDiscardCount(), is(0));
+                            assertThat(getLeaderConfigMap().getData().containsKey(key), is(true));
+
+                            // Second retry should succeed and remove the underlying state and the
+                            // reference in config map.
+                            assertThat(store.releaseAndTryRemove(key), is(true));
+                            assertThat(store.getAllAndLock().size(), is(0));
+                            assertThrows(Exception.class, () -> store.getAndLock(key));
+                            assertThat(TestingLongStateHandleHelper.getGlobalDiscardCount(), is(1));
+                            assertThat(getLeaderConfigMap().getData().containsKey(key), is(false));
+                        });
+            }
+        };
+    }
+
+    @Test
+    public void testReleaseAndTryRemoveAllIsIdempotent() throws Exception {
+        new Context() {
+            {
+                runTest(
+                        () -> {
+                            leaderCallbackGrantLeadership();
+
+                            final KubernetesStateHandleStore<
+                                            TestingLongStateHandleHelper.LongStateHandle>
+                                    store =
+                                            new KubernetesStateHandleStore<>(
+                                                    flinkKubeClient,
+                                                    LEADER_CONFIGMAP_NAME,
+                                                    longStateStorage,
+                                                    filter,
+                                                    LOCK_IDENTITY);
+                            final int numKeys = 10;
+
+                            final RuntimeException discardException =
+                                    new RuntimeException("Test exception.");
+                            for (int idx = 0; idx < numKeys; idx++) {
+                                store.addAndLock(
+                                        key + "_" + idx,
+                                        new TestingLongStateHandleHelper.LongStateHandle(idx + 1) {
```

Comment:
FYI: As already mentioned. I extended `LongStateHandle` to cover this functionality in PR #18869